### PR TITLE
[fix] Removed flattening due to slack webhook only accepts not flatte…

### LINKED ISSUE #11: Sending a message with blocks fails
### Description

Support for advanced notification to slack fails.

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [x] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] example code related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/slack-github-action/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

When using Webhook trigger due to [flatten function](https://github.com/slackapi/slack-github-action/blob/main/index.js#L48 ) action fails is more complex messages.

[flatten(payload)](https://github.com/slackapi/slack-github-action/blob/main/index.js#L48) output:
```
{
  text: 'test',
  'blocks.0.type': 'section',
  'blocks.0.text.type': 'mrkdwn',
  'blocks.0.text.text': 'You have a new request: ',
  'blocks.1.type': 'section',
  'blocks.1.fields.0.type': 'mrkdwn',
  'blocks.1.fields.0.text': '*Type:* Computer (laptop)',
  'blocks.1.fields.1.type': 'mrkdwn',
  'blocks.1.fields.1.text': '*When:* Submitted Aut 10'
}
```

#### Steps to reproduce:
```
- name: Send Slack notification
  uses: slackapi/slack-github-action@v1.15.0
  env:
     SLACK_WEBHOOK_URL:  ***
  with:
     payload: '{"text":"test","blocks":[{"type":"section","text":{"type":"mrkdwn","text":"You have a new request: "}},{"type":"section","fields":[{"type":"mrkdwn","text":"*Type:* Computer (laptop)"},{"type":"mrkdwn","text":"*When:* Submitted Aut 10"}]}]}'
```

#### Expected result:
What you expected to happen
```
**slackBotName**
You have a new request:
Type: Computer (laptop)        When: Submitted Aut 10
```

#### Actual result:

What actually happened
```
**slackBotName**
test
```

### PR DESCRIPTION
**Pull request by m5kr1pka**
###  Summary
Closes #11 Advanced messages fails issue, because slack only accepts not flattened parameters and payload doesn't reach workflow builder afterwards.

### Requirements

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/slack-github-action/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

**Comment by li-olivierfostier** [comment]
Pleaaaase Hurry up ! :)

**Comment by akasakashota** [comment]
Did you forget to build it? I'm looking forward to this PR being merged. Thanks.

**Review by seratch — APPROVED** [review]
LGTM but as I mentioned at https://github.com/slackapi/slack-github-action/issues/11#issuecomment-962688683, I would like to wait for @stevengill's review for this

**Review by seratch — CHANGES_REQUESTED** [review]
Oops, I missed that the current implementation support only Workflow Builder webhook use cases. To support both webhooks for posting a message nad webhooks for triggering a workflow, perhaps, we need to:

* have a new option to determine the webhook type
* dispatch the logic accordingly (=the workflow webhook pattern should be supported too)

**Comment by stevengill** [comment]
Agree with @seratch. I left a comment on the open issue. We don't want to remove flatten from the library but want to introduce a fork for incoming_webhook support.

**Comment by m5kr1pka** [comment]
cc'ing:  @seratch @stevengill 

- Implemented a webhook type via env parameter: `SLACK_WEBHOOK_TYPE`: `BUILDER` or `INCOMING`, by default: `BUILDER` (existing);
- Updated documentation with Incoming webhook;
- added incoming webhook example to example-workflows;
- ran build script and committed unfortunately;
- added @vercel/ncc to devDependencies;

(phew)

P. S.
Would be great to implement tests in this action, but major release would be required

File: index.js [hunk 1]
```diff
@@ -4,16 +4,23 @@ const { WebClient } = require('@slack/web-api');
 const flatten = require('flat');
 const axios = require('axios');
 
+const SLACK_WEBHOOK_TYPES = {
+    WORKFLOW_BUILDER: 'BUILDER',
+    WORKFLOW_INCOMING: 'INCOMING'
```
Comment (seratch) [inline index.js:9]: As the "Incoming Webhooks" feature is not part of workflows / workflow builder, `WORKFLOW_INCOMING` does not make sense. Also, the value "BUILDER" can be confusing. How about renaming this way? Thoughts? @stevengill 
```suggestion
    WORKFLOW_TRIGGER: 'WORKFLOW_TRIGGER',
    INCOMING_WEBHOOK: 'INCOMING_WEBHOOK'
```

File: index.js [hunk 2]
```diff
@@ -4,16 +4,23 @@ const { WebClient } = require('@slack/web-api');
...
+}
+
 try {
     const botToken = process.env.SLACK_BOT_TOKEN;
     const webhookUrl = process.env.SLACK_WEBHOOK_URL;
+    const webhookType = process.env.SLACK_WEBHOOK_TYPE?.toUpperCase() || SLACK_WEBHOOK_TYPES.WORKFLOW_BUILDER;
```
Comment (seratch) [inline index.js:15]:
```suggestion
    // The default type is for Workflow Builder triggers. If you want to use this action for Incoming Webhooks, use the corresponding type instead.
    const webhookType = process.env.SLACK_WEBHOOK_TYPE?.toUpperCase() || SLACK_WEBHOOK_TYPES. WORKFLOW_TRIGGER;
```

File: README.md [hunk 3]
```diff
@@ -66,6 +66,31 @@ This route allows your GitHub Actions job to post a message in a Slack channel o
     SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
 ```
 
+## Slack Incoming Webhook
```
Comment (seratch) [inline README.md:69]: Let's have "Route" for consistency

```suggestion
## Slack Incoming Webhook Route
```

File: README.md [hunk 4]
```diff
@@ -66,6 +66,31 @@ This route allows your GitHub Actions job to post a message in a Slack channel o
...
+
+This route allows your GitHub Actions job to post a message in a Slack channel or direct message by utilizing [messaging.webhooks](https://api.slack.com/messaging/webhooks) API method. With this route, you can instantly post a message.
```
Comment (seratch) [inline README.md:71]:
```suggestion
This route allows your GitHub Actions job to post a message in a Slack channel or direct message by utilizing [Incoming Webhooks](https://api.slack.com/messaging/webhooks). With this route, you can instantly post a message.
```

File: README.md [hunk 5]
```diff
@@ -66,6 +66,31 @@ This route allows your GitHub Actions job to post a message in a Slack channel o
...
+
+Incoming Webhooks conform to the same rules and functionality as any of our other messaging APIs. You can make your posted messages as simple as a single line of text, or make them really useful with [interactive components](https://api.slack.com/messaging/interactivity). To make the message more expressive and useful use [Block Kit](https://api.slack.com/block-kit) to build and test visual components.
+
+## Setup
+
+* [Create a Slack App](https://api.slack.com/apps) for your workspace (alternatively use an existing app you have already created and installed)
+* Install the app to your workspace
+* Invite the bot user into the channel (`/invite @bot_user_name`)
```
Comment (seratch) [inline README.md:79]:
```suggestion
* Add the [incoming-webhook](https://api.slack.com/scopes/incoming-webhook) bot scope under **OAuth & Permissions**
* Install the app to your workspace (you will select a channel to notify)
```

File: README.md [hunk 6]
```diff
@@ -66,6 +66,31 @@ This route allows your GitHub Actions job to post a message in a Slack channel o
...
+
+```
+- name: Send custom JSON data to Slack workflow
+  id: slack
+  uses: slackapi/slack-github-action@v1.15.0
+  with:
+    payload: "{\"blocks\":[{\"type\":\"section\",\"text\":{\"type\":\"mrkdwn\",\"text\":\"You have a new request: \"}},{\"type\":\"section\",\"fields\":[{\"type\":\"mrkdwn\",\"text\":\"*Type:* Computer (laptop)\"},{\"type\":\"mrkdwn\",\"text\":\"*When:* Submitted Aut 10\"}]}]}"
+  env:
+    SLACK_WEBHOOK_URL: ${{ secrets.SLACK_BOT_TOKEN }}
+    SLACK_WEBHOOK_TYPE: INCOMING
```
Comment (seratch) [inline README.md:91]:
```suggestion
    SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
    SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
```

**Review by seratch — CHANGES_REQUESTED** [review]
A few comments. Please wait for @stevengill's review before starting to modify this PR

File: index.js (hunk 2 above)
Comment (m5kr1pka) [reply index.js:15]: Note: whitespace in variable: `SLACK_WEBHOOK_TYPES. WORKFLOW_TRIGGER`

**Comment by m5kr1pka** [comment]
@seratch changes look good to me, thanks for the review, waiting for confirmation.

File: index.js (hunk 2 above)
Comment (seratch) [reply index.js:15]: Thanks, fixed.

File: example-workflows/Incoming webhook/main.yml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    name: New push to repo
+    steps:
+    - name: Send GitHub trigger payload to Slack Workflow Builder
+      id: slack
+      uses: slackapi/slack-github-action@v1.15.0
+      with:
+        payload: "{\"text\":\"Danny Torrence left a 1 star review for your property.\",\"blocks\":[{\"type\":\"section\",\"text\":{\"type\":\"mrkdwn\",\"text\":\"Danny Torrence left the following review for your property:\"}},{\"type\":\"section\",\"block_id\":\"section567\",\"text\":{\"type\":\"mrkdwn\",\"text\":\"<https://example.com|Overlook Hotel> \\n :star: \\n Doors had too many axe holes, guest in room 237 was far too rowdy, whole place felt stuck in the 1920s.\"},\"accessory\":{\"type\":\"image\",\"image_url\":\"https://is5-ssl.mzstatic.com/image/thumb/Purple3/v4/d3/72/5c/d3725c8f-c642-5d69-1904-aa36e4297885/source/256x256bb.jpg\",\"alt_text\":\"Haunted hotel image\"}},{\"type\":\"section\",\"block_id\":\"section789\",\"fields\":[{\"type\":\"mrkdwn\",\"text\":\"*Average Rating*\\n1.0\"}]}]}"
+      env:
+        SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+        SLACK_WEBHOOK_TYPE: incoming
```
Comment (stevengill) [inline example-workflows/Incoming webhook/main.yml:15]:
```suggestion
        SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
```

File: index.js (hunk 1 above)
Comment (stevengill) [reply index.js:9]: I agree @seratch

File: index.js [hunk 8]
```diff
@@ -46,28 +54,37 @@ try {
             payload = github.context.payload;
         }
 
-        // flatten JSON payload (no nested attributes)
-        const flatPayload = flatten(payload);
+        if (webhookType === SLACK_WEBHOOK_TYPES.WORKFLOW_BUILDER) {
```
Comment (stevengill) [inline index.js:57]:
```suggestion
        if (webhookType === SLACK_WEBHOOK_TYPES.WORKFLOW_TRIGGER) {
```

**Review by stevengill — APPROVED** [review]
LGTM

**Pull request merged**